src/mono/mono/mini/aot-compiler.c

```diff
@@ -9832,7 +9856,7 @@ compile_method (MonoAotCompile *acfg, MonoMethod *method)
 	mono_atomic_inc_i32 (&acfg->stats.ccount);
 
 	if (acfg->aot_opts.compiled_methods_outfile && acfg->compiled_methods_outfile != NULL) {
-		if (!mono_method_is_generic_impl (method) && method->token != 0 && !cfg->deopt && !cfg->interp_entry_only) {
```

Comment:
Is the output from this file used to detect what methods should be trimmed ? This looks a bit misleading to me that we don't output here a compiled method (even thought we compiled it) just because interpreter can't call into it. I think the information on whether the interpreter doesn't support calling into a compiled method should be included separately somewhere.

Reply:
Yes, this output file contains the methods could potentially be trimmed. You have a good point. I could emit two files here. One is a complete list of compiled methods. The other is a list of eligible methods for trimming. This might be a overkill for AOT failing back to JIT also. There are more work to do to fine tune it. Will create a task for me to work on it during .NET9.

Reply:
Updated :)